### PR TITLE
dartsim 6.10.1

### PR DESCRIPTION
**Pull request by chenrui333**
[`action-homebrew-bump-formula`](https://github.com/dawidd6/action-homebrew-bump-formula)

---

Created with `brew bump-formula-pr`.

### CODE REVIEW

**Review by dawidd6 — DISMISSED** [review]

File: Formula/dartsim.rb
Comment (jslee02) [inline Formula/dartsim.rb:5]: The tag `v6.10.1` was recently changed. The new sha256 for the new tag is `bf19cdef8e28dbc4059dcbb11997576d6e1d825791bd756e8272d2ddc5b147ce`

Comment (chenrui333) [reply Formula/dartsim.rb]: 👍

**Comment by scpeters** [comment]
similar to the macOS 11 build failure with 6.10.0 in https://github.com/Homebrew/homebrew-core/pull/74912#issuecomment-817202823

~~~
make[2]: *** No rule to make target `/System/Library/Frameworks/GLUT.framework/GLUT', needed by `lib/libdart-gui.6.10.1.dylib'.  Stop.
make[2]: *** Waiting for unfinished jobs....
~~~


> https://github.com/Homebrew/homebrew-core/pull/74912#issuecomment-817367665 by @jslee02 It seems `/System/Library/Frameworks/GLUT.framework/GLUT` is not valid on macOS 11. Would it be possible to make [these lines](https://github.com/Homebrew/homebrew-core/blame/master/Formula/dartsim.rb#L34-L36) conditional to the macOS version? I don't currently have macOS 11 machine to test with this though.

> https://github.com/Homebrew/homebrew-core/pull/74912#issuecomment-817534976 by @SMillerDev Maybe there is a way to prevent dartsim from trying to link agains X11/XQuartz at all?

**Comment by scpeters** [comment]
> similar to the macOS 11 build failure with 6.10.0 in [#74912 (comment)](https://github.com/Homebrew/homebrew-core/pull/74912#issuecomment-817202823)
> 
> ```
> make[2]: *** No rule to make target `/System/Library/Frameworks/GLUT.framework/GLUT', needed by `lib/libdart-gui.6.10.1.dylib'.  Stop.
> make[2]: *** Waiting for unfinished jobs....
> ```
> 
> > [#74912 (comment)](https://github.com/Homebrew/homebrew-core/pull/74912#issuecomment-817367665) by @jslee02 It seems `/System/Library/Frameworks/GLUT.framework/GLUT` is not valid on macOS 11. Would it be possible to make [these lines](https://github.com/Homebrew/homebrew-core/blame/master/Formula/dartsim.rb#L34-L36) conditional to the macOS version? I don't currently have macOS 11 machine to test with this though.
> 
> > [#74912 (comment)](https://github.com/Homebrew/homebrew-core/pull/74912#issuecomment-817534976) by @SMillerDev Maybe there is a way to prevent dartsim from trying to link agains X11/XQuartz at all?

I just looked on a Big Sur machine, and I see the `GLUT.framework` at the following path: `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/SystemLibrary/Frameworks/GLUT.framework/`

I expect there is a helper function that would assist with generating this path for Big Sur only?

**Comment by scpeters** [comment]
> I just looked on a Big Sur machine, and I see the `GLUT.framework` at the following path: `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/SystemLibrary/Frameworks/GLUT.framework/`
> 
> I expect there is a helper function that would assist with generating this path for Big Sur only?

a hint from `jasper`:

https://github.com/Homebrew/homebrew-core/blob/master/Formula/jasper.rb#L27-L32

**Comment by scpeters** [comment]
> > I just looked on a Big Sur machine, and I see the `GLUT.framework` at the following path: `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk/SystemLibrary/Frameworks/GLUT.framework/`
> > I expect there is a helper function that would assist with generating this path for Big Sur only?
> 
> a hint from `jasper`:
> 
> https://github.com/Homebrew/homebrew-core/blob/master/Formula/jasper.rb#L27-L32

I just pushed this approach from `jasper` to this formula in 5c45b6c

**Comment by scpeters** [comment]
this looks good to me, but I'll leave it to another maintainer to approve since I pushed to this branch

**Review by carlocab — APPROVED** [review]

**Pull request closed without merging**
